mountpoint-s3/src/bin/file-system-benchmarks.rs

```diff
@@ -0,0 +1,219 @@
+
+        // Perform and time the open operation
+        let mut file = {
+            let mut open = OpenOptions::new();
+            open.create(true);
+            open.truncate(true);
+            open.write(true);
+            open.read(true);
+
+            let start = Instant::now();
+            let file = open
+                .open(path.clone())
+                .map_err(|e| anyhow::anyhow!("Failed to open file {}: {}", path, e))?;
+            let elapsed_ms = start.elapsed().as_millis_f64_temp();
+            open_latency_samples.push(elapsed_ms);
+            elapsed_total_ms += elapsed_ms;
+            file
+        };
+
+        // Perform and time the writing operation
+        {
+            let start = Instant::now();
+            file.write_all(&vec![0u8; file_size_bytes as usize])
+                .map_err(|e| anyhow::anyhow!("Failed to write to file {}: {}", path, e))?;
+            let elapsed_ms = start.elapsed().as_millis_f64_temp();
+            write_latency_samples.push(elapsed_ms);
+            elapsed_total_ms += elapsed_ms;
+        };
+
+        // Perform and time the flush operation
+        {
+            let start = Instant::now();
+            drop(file);
+            let elapsed_ms = start.elapsed().as_millis_f64_temp();
+            flush_latency_samples.push(elapsed_ms);
+            elapsed_total_ms += elapsed_ms;
+        };
+
+        total_latency_samples.push(elapsed_total_ms);
+
+        fs::remove_file(path.clone()).map_err(|e| anyhow::anyhow!("Failed to remove file {}: {}", path, e))?;
+    }
+
+    let total_latency_result = BenchmarkResult {
+        name: format!("{benchmark_name} - Average Total Latency"),
+        value: mean(&total_latency_samples),
+        unit: Unit::Milliseconds,
+    };
+
+    if !include_breakdown {
+        Ok(vec![total_latency_result])
+    } else {
+        Ok(vec![
+            total_latency_result,
+            BenchmarkResult {
+                name: format!("{benchmark_name} - Average Open Latency"),
+                value: mean(&open_latency_samples),
+                unit: Unit::Milliseconds,
+            },
+            BenchmarkResult {
+                name: format!("{benchmark_name} - Average Write Latency"),
+                value: mean(&write_latency_samples),
+                unit: Unit::Milliseconds,
+            },
+            BenchmarkResult {
+                name: format!("{benchmark_name} - Average Flush Latency"),
+                value: mean(&flush_latency_samples),
+                unit: Unit::Milliseconds,
+            },
+        ])
+    }
+}
+
+fn main() -> Result<()> {
+    let CliArgs {
+        mount_dir,
+        out_file,
+        benchmark_type,
+        detailed,
+    } = CliArgs::parse();
+    let mount_dir_os_str = mount_dir.into_os_string();
+    let mount_dir_str = mount_dir_os_str
+        .to_str()
+        .ok_or_else(|| anyhow!("Invalid UTF-8 in mount directory path"))?;
```

Comment:
Really I think we should have just passed through the `&Path`. We can do things like [`Path::join`](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.join) to get new paths as you use elsewhere (instead of using `format!(..)`.

Reply:
Much better, done!